src/makeCalculator.js

```diff
@@ -3,8 +3,47 @@
 /**
  * @return {object}
  */
+
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+    add(value) {
+      calculator.result += value;
+
+      return calculator;
+    },
+    subtract(value) {
+      calculator.result -= value;
+
+      return calculator;
+    },
+    multiply(value) {
+      calculator.result *= value;
+
+      return calculator;
+    },
+    divide(value) {
+      if (value !== 0) {
+        calculator.result /= value;
+      } else {
+        throw new Error('Cannot divide by zero');
+      }
+
+      return calculator;
+    },
+    reset() {
+      calculator.result = 0;
+
+      return calculator;
+    },
+    operate(fn, value) {
+      fn.call(calculator, value);
+
+      return calculator;
```

Comment:
The `operate` method allows executing any function with the calculator as its context, which can be powerful but also risky. Ensure that the function passed to `operate` is safe and does not modify the calculator state in unintended ways. Consider adding validation or constraints to the function parameter if necessary.